Amazon.QLDB.Driver.Tests/ResultTests.cs

```diff
@@ -108,5 +151,201 @@ public void TestIonEnumeratorResetIsNotSupported()
+            };
+
+            mockSession.Setup(m => m.FetchPage(It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(fetchPageResult);
+
+            var result = new Result(mockSession.Object, "txnId", executeResult);
+
+            var io = result.GetConsumedIOs();
+            var timing = result.GetTimingInformation();
+            Assert.IsNull(io.ReadIOs);
+            Assert.IsNull(io.WriteIOs);
+            Assert.IsNull(timing.ProcessingTimeMilliseconds);
+
+            var results = result.GetEnumerator();
+            while (results.MoveNext())
+            {
+                // Fetch the next page
+            }
+
+            io = result.GetConsumedIOs();
+            timing = result.GetTimingInformation();
+            Assert.IsNull(io.ReadIOs);
+            Assert.IsNull(io.WriteIOs);
+            Assert.IsNull(timing.ProcessingTimeMilliseconds);
+        }
+
+        [TestMethod]
+        public void TestQueryStatsNullExecuteHasFetch()
+        {
+            Mock<Session> mockSession = new Mock<Session>(null, null, null, null, null);
+            var valueHolder = new ValueHolder
+            {
+                IonBinary = memoryStream,
+                IonText = "ionText"
+            };
+            var valueHolderList = new List<ValueHolder> { valueHolder };
+            var executeResult = new ExecuteStatementResult
+            {
+                FirstPage = new Page
+                {
+                    NextPageToken = "hasNextPage",
+                    Values = valueHolderList
+                }
+            };
+            var fetchPageResult = new FetchPageResult
+            {
+                Page = new Page { NextPageToken = null, Values = valueHolderList },
+                ConsumedIOs = fetchIO,
+                TimingInformation = fetchTiming
+            };
+
+            mockSession.Setup(m => m.FetchPage(It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(fetchPageResult);
+
+            var result = new Result(mockSession.Object, "txnId", executeResult);
+
+            var io = result.GetConsumedIOs();
+            var timing = result.GetTimingInformation();
+            Assert.IsNull(io.ReadIOs);
+            Assert.IsNull(io.WriteIOs);
+            Assert.IsNull(timing.ProcessingTimeMilliseconds);
+
+            var results = result.GetEnumerator();
+            while (results.MoveNext())
+            {
+                // Fetch the next page
+            }
+
+            io = result.GetConsumedIOs();
+            timing = result.GetTimingInformation();
+            Assert.AreEqual(fetchReads, io.ReadIOs);
+            Assert.AreEqual(fetchWrites, io.WriteIOs);
+            Assert.AreEqual(fetchTime, timing.ProcessingTimeMilliseconds);
+        }
+
+        [TestMethod]
+        public void TestQueryStatsHasExecuteNullFetch()
+        {
+            Mock<Session> mockSession = new Mock<Session>(null, null, null, null, null);
```

Comment:
There seems to be a lot of duplication between tests. Can we clean that up please?